src/cli.ts

```diff
@@ -102,26 +111,42 @@ for (const file of files) {
     const convertEncoding = new Transform({
         transform(chunk, _encoding, callback) {
             try {
-                const decoded = new TextDecoder('utf-16le').decode(chunk);
-                callback(null,  new TextEncoder().encode(decoded));
+                // Try UTF-16LE first, then fallback to UTF-8 if that fails
+                let decoded;
+                try {
+                    decoded = new TextDecoder('utf-16le').decode(chunk);
+                } catch (utf16Error) {
+                    // If UTF-16LE fails, try UTF-8
+                    decoded = new TextDecoder('utf-8').decode(chunk);
+                }
+                
+                // Clean any invalid characters that might cause XML parsing issues
+                const cleaned = decoded.replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x9F]/g, '');
+                
+                callback(null, new TextEncoder().encode(cleaned));
             } catch (err) {
                 callback(err as Error);
             }
         }
     });
 
+    const saxmlFile = path.join('SaXML', `${file}.xml`);
     await pipeline(
         containerResponse.buffer.stream(),
         convertEncoding,
-        createWriteStream(path.join('SaXML', `${file}.xml`)),
+        createWriteStream(saxmlFile),
     );
 
     log('finished downloading container field', file);
 
-    const clearXml : Partial<FieldData> = {};
+    const clearXml: Partial<FieldData> = {};
     clearXml[field] = '';
 
     await layout.update(Number(records.data[0].recordId), clearXml);
     await client.clearToken();
+
+    await createIdNameMaps(saxmlFile, path.join('stylesheets', `field-map.xsl`), fieldMapFile, isFirstFile);
```

Comment:
When this is installed as a package this paths.join won't work.  you'll need to use import.meta.url fileURLToPath and dirname twice to get the directry for src/build and then dirname again to get the folder stylesheets is in.